data_structures/singly_linked_list.py

```diff
@@ -23,26 +17,24 @@ def __init__(self, li):
         p = q
       p.succ = None
     else:
-      self.head = Node(None)
```

Comment:
Be careful: This is making a list of one element, None, instead of making en empty list.